src/utils/nonce.ts

```diff
@@ -1,45 +1,76 @@
-
 import { defaultRegistryTypes, fromBase64 } from "../stargate";
-import { StdSignature, encodeSecp256k1Pubkey, decodeSignature, coins, pubkeyToAddress } from "../amino";
-import { EncodeObject, makeAuthInfoBytes, makeSignBytes, makeSignDoc, Registry, TxBodyEncodeObject } from "@cosmjs/proto-signing";
+import {
+  StdSignature,
+  encodeSecp256k1Pubkey,
+  decodeSignature,
+  coins,
+  pubkeyToAddress,
+} from "../amino";
+import {
+  EncodeObject,
+  makeAuthInfoBytes,
+  makeSignBytes,
+  makeSignDoc,
+  Registry,
+  TxBodyEncodeObject,
+} from "@cosmjs/proto-signing";
 import { Int53 } from "../math";
 import { Secp256k1, Secp256k1Signature, sha256 } from "../crypto";
 
-export async function verifyNonceMsgSigner(signature: StdSignature, signerAddress: string, nonce: number, sequence: number, accountNumber: number, chainId: string): Promise<boolean> {
-    const txBody = encodeNonce(nonce);
-    const bodyBytes = new Registry(defaultRegistryTypes).encode(txBody);
+export async function verifyNonceMsgSigner(
```

Comment:
can we rename this function to verifyArbitraryMessageSigner for example. the same for sign function